gwsurrogate/surrogateIO.py

```diff
@@ -464,40 +520,102 @@ def amp_fit_func(coeffs,xvec):
+        
       print("Special case: using spline for parametric model at each EI node")
-
-      n_spline_knots = self.file[subdir+'n_spline_knots'][:]
-      spline_knots = self.file[subdir+'spline_knots'][:]
-
-      # pack necessary data up so the spline function can be called
-      # as self.amp_fit_func(self.fitparams_amp[jj,:], x_0)
-      num_fits = self.fitparams_amp.shape[0]
-      fitparams_amp = []
-      fitparams_phase = []
-      degree = int(self.file[subdir+'degree'][:]) # must be int for scipy (> 1.5.2) splev to work
-      for i in range(num_fits):
-        fitparams_amp.append([spline_knots, self.fitparams_amp[i,:], degree])
-        fitparams_phase.append([spline_knots, self.fitparams_phase[i,:], degree])
-      self.fitparams_amp = np.array(fitparams_amp)
-      self.fitparams_phase = np.array(fitparams_phase)
-
-      print("spline knots = %i, num_fits = %i"%(n_spline_knots,num_fits))
-
-      self.amp_fit_func   = my_funcs[self.fit_type_amp]
-      self.phase_fit_func = my_funcs[self.fit_type_phase]
-
+        
+      ## NOTE :We need two if statements for the BHPTNRSur1dq1e4 because, the number of knots are not same
+      ## in different modes or even for different data pieces within the mode. This is because of the
+      ## smoothing we do in fitting. This demands some parts of the code to be re-written.
+      ## Furtermore, we use different decomposition for 22 mode and higher order modes.
+      ## This should be done differently
+      ## =====================================================================================
+      if self.surrogateID=="BHPTNRSur1dq1e4":
+        if self.surrogate_mode_type  == 'amp_phase_basis': # BHPTNRSur1dq1e4's 22 mode is of type amp_phase_basis
+          n_spline_knots_amp = self.file[subdir+'n_spline_knots_amp'][:]
+          n_spline_knots_phase = self.file[subdir+'n_spline_knots_phase'][:]
+
+          spline_knots_amp = self.file[subdir+'spline_knots_amp'][:]
+          spline_knots_phase = self.file[subdir+'spline_knots_phase'][:]
+
+          # pack necessary data up so the spline function can be called
+          # as self.amp_fit_func(self.fitparams_amp[jj,:], x_0)
+          num_fits_amp = self.fitparams_amp.shape[0]
+          num_fits_phase = self.fitparams_phase.shape[0]
+          fitparams_amp = []
+          fitparams_phase = []
+          degree = int(self.file[subdir+'degree'][:]) # must be int for scipy (> 1.5.2) splev to work
+          for i in range(num_fits_amp):
+            fitparams_amp.append([spline_knots_amp[i], self.fitparams_amp[i], degree])
+          for i in range(num_fits_phase):
+            fitparams_phase.append([spline_knots_phase[i], self.fitparams_phase[i], degree])
+          self.fitparams_amp = np.array(fitparams_amp)
+          self.fitparams_phase = np.array(fitparams_phase)
+
+          print("num_fits_amp = %i"%(num_fits_amp))
+          print("num_fits_phase = %i"%(num_fits_phase))
```

Comment:
Why print these? And similar print statements below.

Reply:
This is only when the surrogate is initially loaded. Mostly its just for any visual checks/info/diagnostics. This is standard in the old gws interface